lib/arborist/build-ideal-tree.js

```diff
@@ -810,23 +831,41 @@ This is a one-time fix-up, please be patient...
     }
 
     if (!target) {
-      const current = node.resolve(edge.name)
-      throw Object.assign(new Error('unable to resolve dependency tree'), {
+      const curNode = node.resolve(edge.name)
+      const pc = this[_peerConflict] || { peerConflict: false, current: null }
```

Comment:
is this property `peerConflict` used anywhere?

```suggestion
      const pc = this[_peerConflict] || { current: null }
```

Reply:
Good catch, that should be `{ peer: null, current: null }`, not `peerConflict: false`.